packages/grafana-ui/src/components/Table/Table.tsx

```diff
@@ -1,111 +1,156 @@
+                cell={cell}
+                onCellClick={onCellClick}
+              />
+            ))}
+          </div>
+        );
+      },
+      [prepareRow, rows]
+    );
 
-  return (
-    <div {...getTableProps()} className={tableStyles.table}>
-      <CustomScrollbar hideVerticalTrack={true}>
-        {!noHeader && (
-          <div>
-            {headerGroups.map((headerGroup: HeaderGroup) => (
-              <div className={tableStyles.thead} {...headerGroup.getHeaderGroupProps()} ref={ref}>
-                {headerGroup.headers.map((column: Column, index: number) =>
-                  renderHeaderCell(column, tableStyles.headerCell, data.fields[index])
-                )}
+    return (
+      <div {...getTableProps()} className={tableStyles.table}>
+        <CustomScrollbar hideVerticalTrack={true}>
+          <div style={{ width: `${totalColumnsWidth}px` }}>
+            {!noHeader && (
+              <div>
+                {headerGroups.map((headerGroup: HeaderGroup) => {
+                  return (
+                    <div className={tableStyles.thead} {...headerGroup.getHeaderGroupProps()} ref={ref}>
+                      {headerGroup.headers.map((column: Column, index: number) =>
+                        renderHeaderCell(column, tableStyles, data.fields[index], onColumnResize)
+                      )}
+                    </div>
+                  );
+                })}
               </div>
-            ))}
+            )}
+            <FixedSizeList
+              height={height - headerRowMeasurements.height}
+              itemCount={rows.length}
+              itemSize={tableStyles.rowHeight}
+              width={'100%'}
+              style={{ overflow: 'hidden auto' }}
+            >
+              {RenderRow}
+            </FixedSizeList>
           </div>
-        )}
-        <FixedSizeList
-          height={height - headerRowMeasurements.height}
-          itemCount={rows.length}
-          itemSize={tableStyles.rowHeight}
-          width={totalWidth ?? width}
-          style={{ overflow: 'hidden auto' }}
-        >
-          {RenderRow}
-        </FixedSizeList>
-      </CustomScrollbar>
-    </div>
-  );
-});
+        </CustomScrollbar>
+      </div>
+    );
+  }
+);
 
-function renderHeaderCell(column: any, className: string, field?: Field) {
-  const headerProps = column.getHeaderProps(column.getSortByToggleProps());
-  const fieldTextAlign = getTextAlign(field);
+Table.displayName = 'Table';
 
-  if (fieldTextAlign) {
-    headerProps.style.textAlign = fieldTextAlign;
+function renderHeaderCell(
+  column: any,
+  tableStyles: TableStyles,
+  field?: Field,
+  onColumnResize?: ColumnResizeActionCallback
+) {
+  const headerProps = column.getHeaderProps();
+  if (column.canResize) {
+    headerProps.style.userSelect = column.isResizing ? 'none' : 'auto'; // disables selecting text while resizing
+    if (onColumnResize && field && column.isResizing) {
+      onColumnResize(field, column.width);
```

Comment:
Cannot call callback during rendering, it's what is causing all the errors. 

Looks like you can pass in reducers and listen to the useColumnResizer actions: 
https://github.com/tannerlinsley/react-table/blob/6c2df78b207a1b3a758faa8a63be4e4000c62faa/src/plugin-hooks/useResizeColumns.js#L26

Reply:
Left this callback to be implemented later.